File: Communication entre processus/exo3.1.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>

int main() 
{
    int p[2];
    pipe(p);

    if (fork() == 0) 
    {
        char fd_str[10];
        sprintf(fd_str, "%d", p[0]); 
        close(p[1]); 

        
        char *args[] = {"./fils_exec", fd_str, NULL};
        execv(args[0], args);
        exit(1);
    } 
    else 
    {
        close(p[0]); 
        char *msg = "Message secret du pere";
        write(p[1], msg, strlen(msg) + 1);
        close(p[1]);
        wait(NULL);
    }
    return 0;
}

File: Communication entre processus/exo3.1.py
import os
import sys

def main():
  
    r, w = os.pipe()

    pid = os.fork()

    if pid == 0:
     
        os.close(w) #
        
     
        fd_str = str(r)
        
        # Chemin vers l'interpréteur python et le script fils
        path_to_python = sys.executable
        script_fils = "./fils_exec.py"
        
   
        os.execv(path_to_python, [path_to_python, script_fils, fd_str])
        
      
        os._exit(1)

    else:
        # --- PROCESSUS PARENT ---
        os.close(r)
        
        msg = "Message secret du pere"
        os.write(w, (msg + '\0').encode())
        
        os.close(w)
        os.wait()

if __name__ == "__main__":
    main()